proposals/lip-introduce-events-and-add-events-root-to-block-headers.md

```diff
@@ -0,0 +1,264 @@
+```
+LIP: <LIP number>
+Title: Introduce events and add events root to block headers
+Author:	Mehmet Egemen Albayrak <mehmet.albayrak@lightcurve.io>
+        Alessandro Ricottone <alessandro.ricottone@lightcurve.io>
+Discussions-To: https://research.lisk.com/t/introduce-events-and-add-events-root-to-block-headers/342
+Type: Standards Track
+Created: <YYYY-MM-DD>
+Updated: <YYYY-MM-DD>
+Requires: 0039, 0055
+```
+
+## Abstract
+
+In this LIP, we introduce a mechanism to emit events from the application domain during the block processing. 
+These events are included in a sparse Merkle tree, whose root, the event root, is added as a new property to block headers. 
+Using the event root, it is possible to provide inclusion or or non-inclusion proofs for events, proving whether an event was emitted during the block processing.
+
+## Copyright
+
+This LIP is licensed under the [Creative Commons Zero 1.0 Universal](https://creativecommons.org/publicdomain/zero/1.0/).
+
+## Motivation
+
+Currently in the Lisk protocol, once a transaction has been included in a block, its effect is completely determined by the data inside of it. 
+However, there are several cases in which this may not be the case anymore:
+
+* The result of advanced transactions might depend on the state at the moment of execution, e.g. the conversion rate at which a swap in a DEX is executed, and cannot be read only from the transaction data. 
+* Some state transitions, like assigning the [block reward][lip-0042] to the block generator, are not induced by transactions but [by the block header itself][lip-0055#separation]. 
```

Comment:
This point does not show that the effect of a transaction is not completely determined by its data. It shows that not all state transitions are a result of a transaction. Perhaps remove it as a bullet and add it as a sentence in the paragraph below? so that it further motivates the use of events?

Reply:
good point, added as an extra point after the list